<launch>

<rosparam command="load" file="$(find periodic_slam)/config/simulation.yaml" /> 


    <!-- <param
        name="robot_description"
        command="$(find xacro)/xacro '$(find camera_gazebo)/urdf/moving_cam_pos.urdf.xacro'"
    />
    <include file="$(find camera_gazebo)/launch/hallway.launch">
    </include>
    <node
        name="spawn_model"
        pkg="gazebo_ros"
        type="spawn_model"
        args="-z .5 -urdf -model robot -param robot_description"
    />
 
 
   <node pkg="robot_state_publisher" type="robot_state_publisher"  name="rob_st_pub">
      <param name="publish_frequency" type="double" value="30.0" />
    </node>

    <node pkg="camera_gazebo" type="camera_control_pos.py" name="camera_controller" output="screen">
    </node>  -->

    <!-- <node name ="bag_node" pkg = "rosbag" type = "play" args = "-r .5 /home/hans/Downloads/newCarLargePitch.bag " />  -->
    <!-- <node name ="bag_node" pkg = "rosbag" type = "play" args = "-r .5 /home/hans/downOnlyVIO.bag " />  -->
    <!-- <node name ="bag_node" pkg = "rosbag" type = "play" args = "-r .5 /home/hans/bags/joeBags/indoor1.bag" /> -->
    <!-- <node name ="bag_node" pkg = "rosbag" type = "play" args = "-r 1 /home/hans/bags/joeBags/outdoor4.bag " />  -->
    <!-- <node name ="bag_node" pkg = "rosbag" type = "play" args = "-r 1 /home/hansk/slam_course_ws/slowForward.bag " /> 
     -->
     <!-- <node name="Graphs" pkg="periodic_slam" type="dualGraph" respawn="false" output="screen">
    </node> -->

    <node name="feat" pkg="feature_tracker" type="feature_tracker" respawn="false" output="screen" 
    args=" /home/hans/catkin_ws/src/VINS-Fusion/config/realsense_d435i_sim/realsense_stereo_imu_config.yaml">
    </node>

    <!-- <node name="split" pkg="periodic_slam" type="SplittingFrames" respawn="false" output="screen">
    </node> -->

    <node name="Graph" pkg="periodic_slam" type="stereoISAM2_IMU_VO" respawn="false" output="screen">
    </node>

    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find periodic_slam)/rviz/factor.config.rviz"  />

    <!-- <include file="$(find periodic_slam)/launch/imageProc.launch"/>
 -->

</launch>
